client/src/constants.ts

```diff
@@ -25,6 +25,8 @@ export interface BudgetTransferData {
 
 export type TransactionType = "income" | "expense";
 
+export type BudgetIdType = string | undefined;
```

Comment:
I kept getting an error along the lines of how the data I assigned cannot be assigned to BudgetIdType because it was only string.

As much as I have learned to love TS with this project...it's made me cry a few times

Reply:
Haha I bet.  TS is the way, even though it can be frustrating sometimes!